CMakeLists.txt

```diff
@@ -253,3 +253,81 @@ if (${BOOST_TESTS})
 else()
     MESSAGE("BOOST_TESTS disabled")
 endif()
+
+# Enforce code formatting rules with clang-format.
+find_program(
+    CLANG_FORMAT_PATH
+    NAMES "clang-format"
+    DOC "Path to clang-format executable"
+    )
+if(NOT CLANG_FORMAT_PATH)
+    message(STATUS "clang-format not found.")
+else()
+    message(STATUS "clang-format found: ${CLANG_FORMAT_PATH}")
+    set(DO_CLANG_FORMAT "${CLANG_FORMAT_PATH}" "-i -style=file")
+endif()
+
+# Helper function to make absolute from relative paths.
+function(prepend var prefix)
+    set(listVar "")
+
+    foreach(f ${ARGN})
+        list(APPEND listVar "${prefix}/${f}")
+    endforeach()
+
+    set(${var} "${listVar}" PARENT_SCOPE)
+endfunction()
+
+if(CLANG_FORMAT_PATH)
+    # Gather absolute path to all .cpp files.
```

Comment:
Is there a reason to only apply this to `.cpp` files, should we include `.hpp` files as well?  I think you'd be safe to save some of the complexity here and just recursively scan `kotekan/` and `lib/` for `.cpp|.hpp` files.